draft-jhoyla-req-mtls-flag.md

```diff
@@ -133,7 +133,56 @@ TLS namespace with the following values:
 
 --- back
 
+# Appendix
+
+## Alternative Approaches
+
+### Concealed Auth
+
+Concealed Auth {{?RFC9729}} is a draft that allows a client to attach a header
+at the HTTP layer that cryptographically binds a public key to the underlying
+TLS connection. This gives us a similar authentication property to using
+`request_client_auth`, and is also client initiated. This solution is applicable
+in some contexts, but in the case of authenticating bots is difficult to
+implement.
+
+The way many HTTP libraries work is that they dispatch requests to a connection
+pool. Once the request arrives at the pool it is sent on a chosen connection
+driven by internal logic. This, unfortunately, makes it difficult to know which
+connection a request will be sent on, and thus to produce the appropriate
+signature.
+
+It is possible to avoid this issue, for example by not using connection pooling,
+or by sending a special "pre-request" on each connection that's added to a pool,
+however these solutions tend to be bespoke per library and bring a number of
+their own issues (e.g. handling servers that terminate connections after a
+single request, or handling resumption).
```

Comment:
We should mention the possibility of modifying the pooling code.
```suggestion

It is possible to avoid this issue, for example by not using connection pooling, or modifying
the implementation so that the first request on each new connection includes the binding.
If modifying the implementation is not possible, then there are some ad-hoc options, such as
sending a special "pre-request" on each connection that's added to a pool,
however these solutions tend to be bespoke per library and bring a number of
their own issues (e.g. handling servers that terminate connections after a
single request, or handling resumption).
```